### PR TITLE
Spec: document string literals

### PR DESCRIPTION
**Pull request by laurentlb**
Copied from https://github.com/google/starlark-go/pull/265
I omitted the implementation notes.

### CODE REVIEW

**Comment by laurentlb** [comment]
Note that it doesn't exactly match the Java implementation, e.g. we reject \a.
I assume you'd rather update the implementation?

File: spec.md [hunk 1]
```diff
@@ -299,7 +299,123 @@ octal_digit   = '0' … '7' .
...
+
+An *escaped newline*---that is, a backslash at the end of a line---is ignored,
+allowing a long string to be split across multiple lines of the source file.
+
+```python
+"abc\
+def"			# "abcdef"
+```
+
+An *octal escape* encodes a single byte using its octal value.
```
Comment (alandonovan) [inline spec.md:354]: (The reference to byte here isn't really appropriate for the Java implementation.)

File: spec.md [hunk 2]
```diff
@@ -299,7 +299,123 @@ octal_digit   = '0' … '7' .
...
+It is error if the value is greater than decimal 255.
+
+```python
+'\0'			# "\x00"  a string containing a single NUL byte
+'\12'			# "\n"    octal 12 = decimal 10
+'\101-\132'		# "A-Z"
+'\119'			# "\t9"   = "\11" + "9"
+```
+
+A *hex escape* encodes a single byte using its hexadecimal value.
```
Comment (alandonovan) [inline spec.md:365]: (and hex escapes aren't implemented in Java)

**Review by alandonovan — APPROVED** [review]

**Comment by alandonovan** [comment]
> Note that it doesn't exactly match the Java implementation, e.g. we reject \a.
> I assume you'd rather update the implementation?

Yes, useless though it may be, every language supports \a thus so should we.

File: spec.md (hunk 1 above)
Comment (laurentlb) [reply spec.md:354]: The rest of the spec mentions that a "string represents an immutable sequence of bytes."
I think it's correct that an octal escape encodes a byte.

Do you have another phrasing to suggest?

File: spec.md (hunk 2 above)
Comment (laurentlb) [reply spec.md:365]: Removing

File: spec.md (hunk 1 above)
Comment (alandonovan) [reply spec.md:354]: This seems fine for now. Correctly documenting how the Java implementation differs is a complex and subtle project.

**Pull request merged**